### ISSUE TITLE
Double exponentiation of mask logits

### ISSUE THREAD

**Issue by rubvber** [MichaelKevinKelly/IODINE#1]
I'm also trying to implement/adapt IODINE and trying to clarify some things that weren't totally clear to me from the paper. Your implementation has been very helpful with that - thanks for sharing it!

I did run into one thing that seems wrong to me. In l. 140-141 of iodine.py you do the following:
```
p_mask_individual = (deviation/(2.*self.var_x)).exp().prod(dim=2,keepdim=True) ## (N,K,1,H,W)
p_masks = torch.nn.functional.softmax(p_mask_individual, dim=1) ## (N,K,1,H,W)
```

It seems to me that this (incorrectly) applies the exponential function twice: once explicitly on the first line, and once again as part of the softmax on the second line. Is that a bug or am I missing or misunderstanding something?
